File: server/entities/book_entity.go
package entities

import (
	"time"
	
	"github.com/google/uuid"
)

// BookEntity represents a book in the database layer
type BookEntity struct {
	ID                      int64      `db:"id"`
	Title                   string     `db:"title"`
	Author                  string     `db:"author"`
	Epoch                   *string    `db:"epoch"`
	WordCount               int        `db:"word_count"`
	Embedding               []float64  `db:"embedding"`
	ContentURL              *string    `db:"content_url"`
	Summary                 *string    `db:"summary"`
	Genre                   *string    `db:"genre"`
	DifficultyLevel         int        `db:"difficulty_level"`
	EstimatedReadingMinutes int        `db:"estimated_reading_minutes"`
	DownloadCount           int        `db:"download_count"`
	RatingAverage           float64    `db:"rating_average"`
	RatingCount             int        `db:"rating_count"`
	IsPremium               bool       `db:"is_premium"`
	IsActive                bool       `db:"is_active"`
	CreatedAt               time.Time  `db:"created_at"`
	UpdatedAt               time.Time  `db:"updated_at"`
}

// TableName returns the table name for the entity
func (BookEntity) TableName() string {
	return "books"
}

// ChapterEntity represents a chapter in the database layer
type ChapterEntity struct {
	ID        uuid.UUID `db:"id"`
	BookID    int64     `db:"book_id"`
	Title     string    `db:"title"`
	Content   string    `db:"content"`
	Position  int       `db:"position"`
	WordCount int       `db:"word_count"`
	CreatedAt time.Time `db:"created_at"`
}

// TableName returns the table name for the entity
func (ChapterEntity) TableName() string {
	return "chapters"
}

File: mobile/lib/features/books/domain/entities/book_entity.dart
import 'chapter_entity.dart';

class BookEntity {
  final String id;
  final String title;
  final String author;
  final String? summary;
  final String? genre;
  final String? epoch;
  final int? wordCount;
  final String? contentUrl;
  final String? audioUrl;
  final int? difficultyLevel;
  final int? estimatedReadingMinutes;
  final int downloadCount;
  final double ratingAverage;
  final int ratingCount;
  final bool isPremium;
  final bool isActive;
  final DateTime createdAt;
  final DateTime updatedAt;
  final List<ChapterEntity> chapters;

  BookEntity({
    required this.id,
    required this.title,
    required this.author,
    this.summary,
    this.genre,
    this.epoch,
    this.wordCount,
    this.contentUrl,
    this.audioUrl,
    this.difficultyLevel,
    this.estimatedReadingMinutes,
    this.downloadCount = 0,
    this.ratingAverage = 0.0,
    this.ratingCount = 0,
    this.isPremium = false,
    this.isActive = true,
    required this.createdAt,
    required this.updatedAt,
    this.chapters = const [],
  });

  BookEntity copyWith({
    String? id,
    String? title,
    String? author,
    String? summary,
    String? genre,
    String? epoch,
    int? wordCount,
    String? contentUrl,
    String? audioUrl,
    int? difficultyLevel,
    int? estimatedReadingMinutes,
    int? downloadCount,
    double? ratingAverage,
    int? ratingCount,
    bool? isPremium,
    bool? isActive,
    DateTime? createdAt,
    DateTime? updatedAt,
    List<ChapterEntity>? chapters,
  }) {
    return BookEntity(
      id: id ?? this.id,
      title: title ?? this.title,
      author: author ?? this.author,
      summary: summary ?? this.summary,
      genre: genre ?? this.genre,
      epoch: epoch ?? this.epoch,
      wordCount: wordCount ?? this.wordCount,
      contentUrl: contentUrl ?? this.contentUrl,
      audioUrl: audioUrl ?? this.audioUrl,
      difficultyLevel: difficultyLevel ?? this.difficultyLevel,
      estimatedReadingMinutes: estimatedReadingMinutes ?? this.estimatedReadingMinutes,
      downloadCount: downloadCount ?? this.downloadCount,
      ratingAverage: ratingAverage ?? this.ratingAverage,
      ratingCount: ratingCount ?? this.ratingCount,
      isPremium: isPremium ?? this.isPremium,
      isActive: isActive ?? this.isActive,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
      chapters: chapters ?? this.chapters,
    );
  }

  @override
  String toString() {
    return 'BookEntity(id: $id, title: $title, author: $author, genre: $genre)';
  }

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;
    return other is BookEntity &&
        other.id == id &&
        other.title == title &&
        other.author == author;
  }

  @override
  int get hashCode {
    return id.hashCode ^ title.hashCode ^ author.hashCode;
  }
}